src/lib/dnssd/Types.h

```diff
@@ -204,29 +204,31 @@ inline constexpr size_t kMaxDeviceNameLen         = 32;
 inline constexpr size_t kMaxRotatingIdLen         = 50;
 inline constexpr size_t kMaxPairingInstructionLen = 128;
 
-/// Data that is specific to commisionable/commissioning node discovery
-struct CommissionNodeData
+/// Data that is specific to operational/commisionable/commissioning node discovery
+struct DnssdNodeData
 {
-    size_t rotatingIdLen                                      = 0;
-    uint32_t deviceType                                       = 0;
-    uint16_t longDiscriminator                                = 0;
-    uint16_t vendorId                                         = 0;
-    uint16_t productId                                        = 0;
-    uint16_t pairingHint                                      = 0;
-    uint8_t commissioningMode                                 = 0;
-    uint8_t commissionerPasscode                              = 0;
-    uint8_t rotatingId[kMaxRotatingIdLen]                     = {};
-    char instanceName[Commission::kInstanceNameMaxLength + 1] = {};
-    char deviceName[kMaxDeviceNameLen + 1]                    = {};
-    char pairingInstruction[kMaxPairingInstructionLen + 1]    = {};
-
-    CommissionNodeData() {}
+
+    size_t rotatingIdLen                  = 0;
+    uint32_t deviceType                   = 0;
+    uint16_t longDiscriminator            = 0;
+    uint16_t vendorId                     = 0;
+    uint16_t productId                    = 0;
+    uint16_t pairingHint                  = 0;
+    uint8_t commissioningMode             = 0;
+    uint8_t commissionerPasscode          = 0;
+    uint8_t rotatingId[kMaxRotatingIdLen] = {};
+    bool nodeStatus; // true is ttl > 0 false id ttl = 0
```

Comment:
This needs a better name. how about `hasZeroTTL` (but then condition is inverted) or something similar?

I should not need to read comments to understand what a value means.

Reply:
That makes sense, application can decide how to use the info of ttl being 0. Will update it to `hasZeroTTL`.